File: src/Core/Entities/Recruitment.cs
using SharedKernel.Bases;

namespace Core.Entities;

public class Recruitment : ModifyEntity
{
  public string Name { get; set; } = string.Empty;
  public string Content { get; set; } = string.Empty;
  public string Benifit { get; set; } = string.Empty;
  public DateTime StartDate { get; set; }
  public DateTime EndDate { get; set; }
  public int SalaryMin { get; set; }
  public int SalaryMax { get; set; }
  public int ExperienceFrom { get; set; }
  public int ExperienceTo { get; set; }
  public int Number { get; set; }
  public int PositionId { get; set; }
  public int DepartmentId { get; set; }

  public virtual Department? Department { get; set; }
  public virtual Position? Position { get; set; }
  public virtual ICollection<RecruitSkillTag>? SkillTags { get; set; }

  private readonly List<Candidate> _candidates = new();
  public IReadOnlyCollection<Candidate> Candidates => _candidates.AsReadOnly();

  public void AddCandidate(Candidate candidate) => _candidates.Add(candidate);
}

File: src/ui/src/apis/recruitment.ts
import { Moment } from "moment"
import { GeneralCandidateProps } from "./candidate"
import client, { Response } from "./client"

export interface GeneralRecruitmentProps{
  id?: number
  name: string
  startDate: Moment
  endDate: Moment
  salaryMin: number
  salaryMax: number
  experienceFrom: number
  experienceTo: number
  number: number
}

export type RecruitmentRequest = GeneralRecruitmentProps & {
  content: string
  benifit: string
  departmentId: number
  positionId: number
}

export interface RecruitmentDetailProps extends RecruitmentRequest {
  candidates: Array<GeneralCandidateProps>
}

type RecruitmentsResponse = Response<Array<GeneralRecruitmentProps>>
type RecruitmentDetailResponse = Response<RecruitmentDetailProps>

export const postRecruitment = async (data: RecruitmentRequest): Promise<Response<GeneralRecruitmentProps>> => {
  const response = await client.post('/recruitments', data)
  return response.data
}

export const patchRecruitment = async (id: string, data: RecruitmentRequest): Promise<Response<GeneralRecruitmentProps>> => {
  const response = await client.patch(`/recruitments/${id}`, data)
  return response.data
}

export const getRecruitments = async (): Promise<RecruitmentsResponse> => {
  const response = await client.get('/recruitments')
  return response.data
}

export const getRecruitmentDetail = async (id: string | number): Promise<RecruitmentDetailResponse> => {
  const response = await client.get(`/recruitments/${id}`)
  return response.data
}